### PR TITLE
Fix spelling across the entire repo

### PR DESCRIPTION
**Pull request by BelleNottelling**
Additionally, also created a config file so we can use [cspell](https://cspell.org/) to easily check it again in the future. I intend to do that before each release

### CODE REVIEW

**Review by jaapmarcus — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -403,8 +403,9 @@ Release 2.6.17 (2012-06-17)
 * New dashboard widget to show active orders count grouped by products
 * Updated Twig to 1.8.2 version
 * Added ability to disable/enable automatic invoice issuing for orders
-* SpamChecker is a module. Google recapthca and stopforumspam.com database checks can be simply enabled or disabled as any other module
-* All extensions can call other extension event hooks
+* SpamChecker is a module. Google recaptcha and stopforumspam.com database checks can be simply enabled or disabled as any other module
+* All extensions can call other extension even
+* t hooks
```
Comment (evrifaessa) [inline CHANGELOG.md:408]: ?

File: src/bb-modules/Servicedomain/Api/Guest.php [hunk 2]
```diff
@@ -134,7 +134,7 @@ public function can_be_transferred($data)
         if (!$tld instanceof \Model_Tld) {
             throw new \Box_Exception('TLD is not active.');
         }
-        if (!$this->getService()->canBeTransfered($tld, $data['sld'])) {
+        if (!$this->getService()->canBetransferred($tld, $data['sld'])) {
```
Comment (evrifaessa) [inline src/bb-modules/Servicedomain/Api/Guest.php:137]: ?

**Comment by evrifaessa** [comment]
Does it rename any API routes or event names? We might want to avoid that or document exactly what was changed.

File: src/bb-modules/Servicedomain/Api/Guest.php (hunk 2 above)
Comment (jaapmarcus) [reply src/bb-modules/Servicedomain/Api/Guest.php:137]: What Is the Correct Spelling: Transferred or Transfered? “Transferred” is the only correct way of spelling this verb. “Transfered” is not a word and can't be used in any situation whatsoever.

Comment (evrifaessa) [reply src/bb-modules/Servicedomain/Api/Guest.php:137]: I was talking about the capitalization. I think it should be `canBeTransferred` if "transferred" is correct.

**Comment by BelleNottelling** [comment]
> Does it rename any API routes or event names? We might want to avoid that or document exactly what was changed.

It does not. The only functions that got renamed are internal ones and not API endpoints, and all references have been updated as well

**Comment by John-S4** [comment]
Can't say I'm particularly happy about uses of British English being corrected as errors in favour (apparently spelt wrong) of American spellings. 

But hey, there were a huge amount of actual typos and errors that it caught.

**Comment by jaapmarcus** [comment]
We are probally able to set it to British or American English but we should decide one time. And make it consistent..

**Comment by BelleNottelling** [comment]
> Can't say I'm particularly happy about uses of British English being corrected as errors in favour (apparently spelt wrong) of American spellings. 
> 
> But hey, there were a huge amount of actual typos and errors that it caught. 

Those.. were actually me. It flagged them as typos and I changed them without even thinking about it tbh. It's been a long day.

Feel free to edit the PR or let me know which ones were wrongly changed and I'll correct it tomorrow (and add them to the word list)

**Review by John-S4 — APPROVED** [review]

**Pull request merged**